service/history/workflow/context.go

```diff
@@ -445,17 +445,14 @@ func (c *ContextImpl) CreateWorkflowExecution(
 		HistorySize: historySize,
 	}
 
-	_, err := createWorkflowExecutionWithRetry(c.shard, createRequest)
-	if err != nil {
+	if err := createWorkflowExecutionWithRetry(
+		c.shard,
+		createRequest,
+	); err != nil {
 		return err
 	}
 
-	c.notifyTasks(
-		newWorkflow.TransferTasks,
-		newWorkflow.TimerTasks,
-		newWorkflow.ReplicationTasks,
-		newWorkflow.VisibilityTasks,
-	)
+	notifyWorkflowSnapshotTasks(c.engine, newWorkflow)
```

Comment:
nit: rename to notifyWorkflowTasks

Reply:
there are 2 versions
one for workflow snapshot
one for workflow mutation
due to type difference